R/utility.R

```diff
@@ -1,35 +1,46 @@
-getInstallationCommand <- function(packages){
-  installation <- ""
-
-  for(package in packages){
-    installation <- paste0(installation,
-                         sprintf("Rscript -e \'args <- commandArgs(TRUE)\' -e \'install.packages(args[1], dependencies=TRUE)\' %s", package),
-                         ";")
+getJobPackageInstallationCommand <- function(type, packages){
+  script <- ""
+  if (type == "cran") {
+    script <- "Rscript $AZ_BATCH_JOB_PREP_WORKING_DIR/install_cran.R"
+  }
+  else if (type == "github") {
```

Comment:
The only issue we have is how to differentiate a github / cran package in the foreach argument

foreach( i = 1:3, .packages =c("httr", "lubridate"), .github = c("Azure/doAzureParallel")) {}